CRM/Contribute/BAO/Contribution/Utils.php

```diff
@@ -231,7 +231,7 @@ public static function processConfirm(
    * @return bool
    */
   static protected function isPaymentTransaction($form) {
-    if (!empty($form->_values['is_monetary']) && $form->_amount >= 0.0) {
+    if ($form->_amount == 0.0 || (!empty($form->_values['is_monetary']) && $form->_amount > 0.0)) {
```

Comment:
In other words it could just be if ($form->_amount == 0.0) ?

(Which makes sense to me - I have a deep seated hatred of the is_monetary checkbox as I think it is otherwise calculable & we should actively deprecate it.)